### PR TITLE
Refactor Raspi launcher

### PR DESCRIPTION
**Pull request by kaidokert**
Series of commits to bolster reliability of Raspi launcher. 
- First adding basic unit/mock tests to the existing launcher code
- Add a separate module with retry handling
- Refactor existing code to use the modular retry handler, no functional changes
- Add more retries and guard failure paths to improve the reliability

b/240984469
b/278447902

### CODE REVIEW

**Comment by kaidokert** [comment]
This is ready for a first round of review. I'm still re-running a few on-device test builds and may need to add some more minor tweak in a new commit.

**Review by oxve — APPROVED** [review]
[And if at first you don't succeed
Then dust yourself off and try again](https://www.youtube.com/watch?v=qTA0RuZoIxM)

File: starboard/raspi/shared/launcher_test.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  def test_run(self):
+    result = self._make_launcher().Run()
+    # Expect test failure
+    self.assertEqual(result, 1)
+
+  def test_ip(self):
+    self.assertEqual(self._make_launcher().GetDeviceIp(), self.device_id)
+
+  def test_output(self):
+    self.assertEqual(self._make_launcher().GetDeviceOutputPath(), '/tmp')
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply starboard/raspi/shared/launcher_test.py:83]: It's not, that `/tmp` is hardcoded into the launcher itself actually.

**Comment by kaidokert** [comment]
To be clear this entire launcher needs to be fully reimplemented ( either on top of Fabric or Paramiko+Invoke ) as `pexpect` will never be properly reliable for this use.

I have a quick draft PR here that seems to work: https://github.com/youtube/cobalt/pull/112/files 

However, we need to solve our Python dependencies bundling before we can do that ( b/279249837 )

File: starboard/raspi/shared/launcher.py [hunk 2]
```diff
@@ -255,7 +257,7 @@ def _CleanupPexpectProcess(self):
 
       # Send ctrl-c to the raspi and close the process.
       self._PexpectSendLine(chr(3))
-      time.sleep(1)  # Allow a second for normal shutdown
+      time.sleep(self._PEXPECT_TIMEOUT)  # Allow a second for normal shutdown
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply starboard/raspi/shared/launcher.py]: Modified comment

**Comment by kaidokert** [comment]
Both last turn build failures were raspi-2, but neither was this `launcher.py` problem

**Pull request merged**